File: 2634-minimum-common-value/minimum-common-value.py
class Solution:
    def getCommon(self, nums1: list[int], nums2: list[int]) -> int:
        n1=len(nums1);n2=len(nums2)
        l=r=0
        while l<n1 and r<n2:
            if nums1[l]==nums2[r]:
                return nums1[l]
            elif nums1[l]<nums2[r]:
                l+=1
            else:
                r+=1
        return -1

File: 2634-minimum-common-value/minimum-common-value.java
class Solution {
    public int getCommon(int[] nums1, int[] nums2) {
        int n1=nums1.length, n2=nums2.length;
        int l=0, r=0;
        while(l<n1 & r<n2){
            if(nums1[l]==nums2[r])  return nums1[l];
            else if(nums1[l]>nums2[r])  r+=1;
            else    l+=1;
        }
        return -1;
    }
}